### PR TITLE
docs(nakama): Add docs for Nakama and Cardinal

### PR DESCRIPTION
**Pull request by jerargus**
## What is the purpose of the change

Add docs that describe Nakama features and how those features are used in the starter-game-template to make a general purpose Cardinal plugin for Nakama.

### CODE REVIEW

File: docs/pages/Cardinal/ecs.mdx [hunk 1]
```diff
@@ -20,4 +20,4 @@ Systems are the heartbeat of ECS, encompassing functions that manipulate entitie
 
 ## Why Do We Need ECS?
 
-When used correctly, ECS unlocks a realm of organized code and seamless extensibility. Injecting new features into your game can be as straightforward as introducing a novel system that interacts with existing components. It's important to note that while ECS is a formidable tool, it isn't a universal solution. Certain game scenarios might find better suited solutions outside the ECS paradigm.
\ No newline at end of file
+When used correctly, ECS unlocks a realm of organized code and seamless extensibility. Injecting new features into your game can be as straightforward as introducing a novel system that interacts with existing components. It's important to note that while ECS is a formidable tool, it isn't a universal solution. Certain game scenarios might find better suited solutions outside the ECS paradigm.
```
Comment (pyrofolium) [inline docs/pages/Cardinal/ecs.mdx:23]: Maybe add links where the user can go to explore more about the ECS pattern.

File: docs/pages/Nakama/plugin.mdx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+```go
+var MoveTx = cardinal.NewTransactionType[MoveMsg, MoveResult]("move-a-player")
+var PositionRead = cardinal.NewReadType[PositionRequest, PositionReply]("position")
+```
+
+And register those objects with `world.RegisterTransactions` and `world.RegisterReads`, the following RPC endpoints will be created in Nakama:
+
+- /tx/game/move-a-player
+- /read/game/position
+
```
Comment (pyrofolium) [inline docs/pages/Nakama/plugin.mdx:20]: Are these the new endpoints? I would personally still leave the old ones there as it's still usable. The new endpoints are sort of rushed while the old ones are more stable and hardened for now. Just my opinion. 

Also the /read/game/position has been replaced with /query/game/position.

**Review by pyrofolium — APPROVED** [review]
LGTM

File: docs/pages/Nakama/plugin.mdx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+RPC requests to those Nakama endpoints will be passed along to your Cardinal implementation. Transactions will be added to the transaction queue (which can be processed at the next game tick in a registered System), and Reads will be passed directly to Cardinal.
+
+The communication from a client (e.g. the Nakama console or a custom-built game client) to Nakama uses the RPC protocol. Communication between Nakama and Cardinal makes use of HTTP. Request payloads and response payloads are JSON encoded strings.
+
+## Transaction Signatures
+
+When submitting transactions from Nakama to Cardinal, transactions must be cryptographically signed. The Cardinal plugin will use an existing private key, or generate one if one does not exist. This key generation and signing happens automatically. To make use of this automatic signing, a client must first register a >>>Persona Tag<<<(LINK_TO_MARKDOWN). Once a Persona Tag has been requested and accepted, future transactions will be automatically signed.
```
Comment (smsunarto) [inline docs/pages/Nakama/plugin.mdx:27]: Seems like the >>> <<< thing is causing syntax error, and i think you forgot to add the link to the markdown here

**Review by smsunarto — APPROVED** [review]
I recommend we pause this until the OpenAPI/swagger stuff is in because that's going to change the endpoints a lot

**Review by smsunarto — CHANGES_REQUESTED** [review]
I recommend we pause this until the OpenAPI/swagger stuff is in because that's going to change the endpoints a lot

**Review by namrapatel — APPROVED** [review]

**Comment by technicallyty** [comment]
title here should be `docs: nakama and cardinal` , this isn't necessarily a feature PR

File: docs/pages/Nakama/plugin.mdx [hunk 4]
```diff
@@ -0,0 +1,201 @@
+# Cardinal Plugin
+
+Go based Nakama containers can be extended using shared objects (aka [plugins](https://pkg.go.dev/plugin)). The [starter-game-template](https://github.com/Argus-Labs/starter-game-template) leverages this shared object pattern to expand on Nakama to better interact with a Cardinal based backend.
```
Comment (technicallyty) [inline docs/pages/Nakama/plugin.mdx:3]:
```suggestion
Go-based Nakama containers can be extended using shared objects (aka [plugins](https://pkg.go.dev/plugin)). The [starter-game-template](https://github.com/Argus-Labs/starter-game-template) leverages this shared object pattern to expand on Nakama to better interact with a Cardinal based backend.
```

File: docs/pages/Nakama/plugin.mdx [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+The Nakama plugin bundled with the starter-game-template is general purpose, and should work with your Cardinal based game without any modifications. As you work on your game, you may find that you want to further change the Nakama plugin to better serve your specific use case. This document serves as a reference for the various aspects of the Cardinal-focused Nakama plugin.
+
+## RPC Endpoints
+
+Transactions and Reads registered with a `cardinal.World` object will automatically get Nakama RPC endpoints created for them. For example, if you set up a Transaction and a Read with the following code:
+
+```go
+var MoveTx = cardinal.NewTransactionType[MoveMsg, MoveResult]("move-a-player")
+var PositionRead = cardinal.NewReadType[PositionRequest, PositionReply]("position")
```
Comment (technicallyty) [inline docs/pages/Nakama/plugin.mdx:13]: maybe this is not important, but there aren't enough args to `NewReadType` here. maybe we can ellipsis to indicate there should be more after "position"

File: docs/pages/Nakama/plugin.mdx [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+```
+
+And register those objects with `world.RegisterTransactions` and `world.RegisterReads`, the following RPC endpoints will be created in Nakama:
+
+- /tx/game/move-a-player
+- /read/game/position
+
+RPC requests to those Nakama endpoints will be passed along to your Cardinal implementation. Transactions will be added to the transaction queue (which can be processed at the next game tick in a registered System), and Reads will be passed directly to Cardinal.
```
Comment (technicallyty) [inline docs/pages/Nakama/plugin.mdx:21]: wdyt of explaining the differences of behavior of responses btwn transactions and reads? i.e. reads reply directly thru the RPC request, txs get streamed thru the match?

File: docs/pages/Nakama/plugin.mdx [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+
+## Transaction Signatures
+
+When submitting transactions from Nakama to Cardinal, transactions must be cryptographically signed. The Cardinal plugin will use an existing private key, or generate one if one does not exist. This key generation and signing happens automatically. To make use of this automatic signing, a client must first register a >>>Persona Tag<<<(LINK_TO_MARKDOWN). Once a Persona Tag has been requested and accepted, future transactions will be automatically signed.
+
+After generation, the key is stored in Nakama's storage layer under the admin User ID.
+
+| Collection Name        | Key Name          | User ID                              | Description                                      |
+|------------------------|-------------------|--------------------------------------|--------------------------------------------------|
+| private_key_collection | private_key_key   | 00000000-0000-0000-0000-000000000000 | The actual private key                           |
```
Comment (technicallyty) [inline docs/pages/Nakama/plugin.mdx:33]: maybe for description here we can describe the type of private key: `ECDSA`

**Comment by pyrofolium** [comment]
One thing I just thought of @jerargus. Mage has certain commands that just don't work and aren't clear about what they do.

I was just informed that something like mage start is actually part of starter-game-template.

File: docs/pages/Nakama/plugin.mdx (hunk 2 above)
Comment (jerargus) [reply docs/pages/Nakama/plugin.mdx:20]: I'll hold off on submitting this PR until the new endpoints are merged.

File: docs/pages/Nakama/plugin.mdx (hunk 3 above)
Comment (jerargus) [reply docs/pages/Nakama/plugin.mdx]: I wasn't sure on the syntax. Hopefully these are fixed now.

File: docs/pages/Nakama/intro.mdx [hunk 8]
```diff
@@ -0,0 +1,14 @@
+# Nakama
+
+[Nakama](https://heroiclabs.com/nakama/) is an open source server framework geared towards multiplayer experiences. It has a number of different client libraries (such as Unity and Javascript) which makes it easy to implement game clients that interact with Nakama.
+
+Go based Nakama deployments can be extended by using "shared objects", also known as [plugins](https://pkg.go.dev/plugin).
```
[the remark this answers is not in the document]
Comment (jerargus) [reply docs/pages/Nakama/intro.mdx]: done

File: docs/pages/Nakama/plugin.mdx (hunk 5 above)
Comment (jerargus) [reply docs/pages/Nakama/plugin.mdx]: I added a placeholder function name that isn't actually defined anywhere. Let me know if you think this should be "..." instead (or maybe a stubbed anonymous function).

File: docs/pages/Nakama/plugin.mdx (hunk 6 above)
Comment (jerargus) [reply docs/pages/Nakama/plugin.mdx]: Done.

File: docs/pages/Nakama/plugin.mdx (hunk 7 above)
Comment (jerargus) [reply docs/pages/Nakama/plugin.mdx]: done

File: docs/pages/Nakama/plugin.mdx (hunk 4 above)
Comment (jerargus) [reply docs/pages/Nakama/plugin.mdx]: done

**Comment by technicallyty** [comment]
we do have some hackers @ interchain HH that might find this doc useful in some capacity. maybe we can merge as-is so they have something to read to better understand how/why we're using nakama, and revisit this later. @smsunarto

**Comment by jerargus** [comment]
> we do have some hackers @ interchain HH that might find this doc useful in some capacity. maybe we can merge as-is so they have something to read to better understand how/why we're using nakama, and revisit this later. @smsunarto 
> 

@technicallyty I added sections for both the current endpoints as well as the endpoints Brian is working on.

File: docs/pages/Cardinal/ecs.mdx (hunk 1 above)
Comment (jerargus) [reply docs/pages/Cardinal/ecs.mdx:23]: This is out of scope for this PR (focusing on Nakama docs).

**Review by technicallyty — APPROVED** [review]

**Review by smsunarto — APPROVED** [review]

**Pull request merged**